gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -0,0 +1,537 @@
+    DESCRIPTION("description"),
+    DNS_NAME("dnsName"),
+    ZONE_ID("id"),
+    NAME("name"),
+    NAME_SERVER_SET("nameServerSet"),
+    NAME_SERVERS("nameServers");
+
+    private final String selector;
+
+    ZoneField(String selector) {
+      this.selector = selector;
+    }
+
+    String selector() {
+      return selector;
+    }
+
+    static String selector(ZoneField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(NAME.selector());
+      for (ZoneField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The fields of a record set.
+   *
+   * <p>These values can be used to specify the fields to include in a partial response when calling
+   * {@link Dns#listRecordSets(String, RecordSetListOption...)}. The name and type are always
+   * returned even if not selected.
+   */
+  enum RecordSetField {
+    DNS_RECORDS("rrdatas"),
+    NAME("name"),
+    TTL("ttl"),
+    TYPE("type");
+
+    private final String selector;
+
+    RecordSetField(String selector) {
+      this.selector = selector;
+    }
+
+    String selector() {
+      return selector;
+    }
+
+    static String selector(RecordSetField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(NAME.selector());
+      fieldStrings.add(TYPE.selector());
+      for (RecordSetField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The fields of a change request.
+   *
+   * <p>These values can be used to specify the fields to include in a partial response when calling
+   * {@link Dns#applyChangeRequest(String, ChangeRequestInfo, ChangeRequestOption...)} The ID is always
+   * returned even if not selected.
+   */
+  enum ChangeRequestField {
+    ID("id"),
+    START_TIME("startTime"),
+    STATUS("status"),
+    ADDITIONS("additions"),
+    DELETIONS("deletions");
+
+    private final String selector;
+
+    ChangeRequestField(String selector) {
+      this.selector = selector;
+    }
+
+    String selector() {
+      return selector;
+    }
+
+    static String selector(ChangeRequestField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(ID.selector());
+      for (ChangeRequestField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The sorting order for listing.
+   */
+  enum SortingOrder {
+    DESCENDING, ASCENDING;
+
+    public String selector() {
+      return name().toLowerCase();
+    }
+  }
+
+  /**
+   * Class for specifying record set listing options.
+   */
+  class RecordSetListOption extends AbstractOption implements Serializable {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://www.codacy.com/app/mziccard/gcloud-java/file/2317493773/issues/source?bid=3123109&fileBranchId=3203698#l185)